### PR TITLE
Cleaning up imageuploader

### PR DESCRIPTION
**Pull request by ViXXoR**
Adding better error handling with description toast message and removing issues causing the user to have to reload the page
Adds change with relation to jellyfin/jellyfin#247

### CODE REVIEW

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 1]
```diff
@@ -1,177 +1,142 @@
...
     function onFileReaderError(evt) {
-
         loading.hide();
-
         switch (evt.target.error.code) {
             case evt.target.error.NOT_FOUND_ERR:
-                require(['toast'], function (toast) {
-                    toast(globalize.translate('sharedcomponents#MessageFileReadError'));
+                require(["toast"], function(toast) {
+                    toast(globalize.translate("sharedcomponents#MessageFileReadError"))
```
Comment (dkanada) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: Why remove semicolons here?

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 2]
```diff
@@ -1,177 +1,142 @@
...
                 });
                 break;
             case evt.target.error.ABORT_ERR:
-                break; // noop
+                break;
             default:
-                require(['toast'], function (toast) {
-                    toast(globalize.translate('sharedcomponents#MessageFileReadError'));
+                require(["toast"], function(toast) {
+                    toast(globalize.translate("sharedcomponents#MessageFileReadError"))
```
Comment (dkanada) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: Same.

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 3]
```diff
@@ -1,177 +1,142 @@
...
-
-        page.querySelector('#uploadImage').addEventListener("change", function () {
+        page.querySelector("form").addEventListener("submit", onSubmit);
+        page.querySelector("#uploadImage").addEventListener("change", function() {
             setFiles(page, this.files);
         });
-
-        page.querySelector('.btnBrowse').addEventListener("click", function () {
-            page.querySelector('#uploadImage').click();
+        page.querySelector(".btnBrowse").addEventListener("click", function() {
```
Comment (dkanada) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: pretty sure we are going with single quotes, #1 was switching them over

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 4]
```diff
@@ -1,177 +1,142 @@
...
                 });
-                break;
         }
     }
 
     function setFiles(page, files) {
-
         var file = files[0];
-
-        if (!file || !file.type.match('image.*')) {
```
Comment (JustAMan) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: Why change this piece at all?

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 5]
```diff
@@ -1,177 +1,142 @@
...
             currentItemId = options.itemId;
             currentServerId = options.serverId;
-
             var dialogOptions = {
                 removeOnClose: true
             };
-
-            if (layoutManager.tv) {
-                dialogOptions.size = 'fullscreen';
+            if(layoutManager.tv) {
```
Comment (JustAMan) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: Why did you remove the space and replaced quotes everywhere?

**Comment by ViXXoR** [comment]
So, just to clear some things up. I did this cleanup because I didn't see any changes to it in #1. It's looking like something that has now been merged into dev has made changes to this file. So because of that it looks like I'm making sweeping changes to already cleaned code, which I did not do.

I pulled the commit over from the other branch, which I had completely tested, and didn't have any invalid JS in it. So, again, something went really wrong between than branch and this one. It's my fault for not doubling checking after moving the commit. However, I feel like it wound't have matter because a lot of this is due to changes in #1.

TBH, the major change needed are the toast messages for errors related to the file type and image type.  So for now I'll remove my changes and redo them against the changes in #1, which, again, I thought were not affecting this file.

All your comments were also pretty accusatory, which is leaving a bit of a bad taste in my mouth. I'll come back and fix up the other fixes I wanted to add to prevent other major issues later.

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 6]
```diff
@@ -1,177 +1,142 @@
...
         reader.readAsDataURL(file);
     }
 
     function onSubmit(e) {
-
+        var dlg = dom.parentWithClass(this, "dialog");
         var file = currentFile;
-
-        if (!file) {
+        if (!file || ("image/png" !== file.type && "image/jpeg" !== file.type && "image/jpeg" !== file.type)) {
```
[the remark this answers is not in the document]
Comment (cvium) [reply src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: #1 isn't exactly a cleaned up version, it only changed the minified files to the proper upstream source, so any weirdness introduced in that PR is old Emby code as well :grinning:

I only commented on it because it stated you'd made changes to it. Did you base your work on an older commit perhaps?

**Comment by dkanada** [comment]
@ViXXoR sorry this happened, I wasn't accusing you at all but was just wondering why a few sections were changed. Please don't think I was trying to be hostile, I was just wondering which sections were applying the fix you moved out of the other pull request.

**Comment by ViXXoR** [comment]
Okay, I've re-implemented my changes on top of the merged #1.

Everything is looking much nicer now, and the change list much smaller.

I'll admit (for lack of better wording at the moment) that my jimmies were a little rustled this morning. I was not expecting so many comments for what I thought were small changes. I overreacted as well as I apologize.

Please take a look and let me know if anything looks out of place.

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 7]
```diff
@@ -74,7 +74,11 @@
             return false;
         }
 
-        if (file.type !== "image/png" && file.type !== "image/jpeg" && file.type !== "image/jpeg") {
+        if (file.type !== "image/png" && file.type !== "image/jpeg") {
+            require(['toast'], function (toast) {
+                toast(globalize.translate('sharedcomponents#MessageImageFileTypeAllowed'));
+            });
+            e.preventDefault();
```
Comment (dkanada) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: This change looks great!

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.template.html [hunk 8]
```diff
@@ -22,12 +22,13 @@ <h2 style="margin:0;">${HeaderAddUpdateImage}</h2>
                 <div class="imageEditor-dropZone fieldDescription">
                     <div>${LabelDropImageHere}</div>
                     <output id="imageOutput" class="flex align-items-center justify-content-center" style="position: absolute;top:0;left:0;right:0;bottom:0;width:100%;"></output>
-                    <input type="file" accept="image/*" id="uploadImage" name="uploadImage" style="position: absolute;top:0;left:0;right:0;bottom:0;width:100%;opacity:0;" />
+                    <input type="file" accept="image/x-png,image/jpeg" id="uploadImage" name="uploadImage" style="position: absolute;top:0;left:0;right:0;bottom:0;width:100%;opacity:0;" />
```
Comment (nvllsvm) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.template.html]: Should be `image/png` since that's the official IANA media type.
https://www.iana.org/assignments/media-types/media-types.xhtml

Comment (ViXXoR) [reply src/bower_components/emby-webcomponents/imageuploader/imageuploader.template.html]: I'll include it, but also leave the x-, as it seems some software may still use that MIME. So I'll just be safe.

File: src/bower_components/emby-webcomponents/imageuploader/imageuploader.js [hunk 9]
```diff
@@ -74,7 +74,11 @@
             return false;
         }
 
-        if (file.type !== "image/png" && file.type !== "image/jpeg" && file.type !== "image/jpeg") {
+        if (file.type !== "image/png" && file.type !== "image/jpeg") {
```
Comment (JustAMan) [inline src/bower_components/emby-webcomponents/imageuploader/imageuploader.js]: Well, if you check for `x-png` below why not check for it here as well?

**Comment by joshuaboniface** [comment]
@nvllsvm I think this one is good to go now as your review was resolved.
